### PR TITLE
feat: extend holochain-turn-server module and add stun-0.main.infra.holo.host instance

### LINKED ISSUE #3479: Run a separate STUN infrastructure server
Run a separate STUN infrastructure server so that it can keep running and connecting folks that are able even if our TURN server is overloaded / offline.

- Pick a STUN server:
  - [coturn](https://github.com/coturn/coturn) venerable, but lightly maintained
  - [eturnal](https://eturnal.net/) fresh but written in erlang
- DISABLE TURN IN THE CONFIG
  - [coturn: stun-only](https://github.com/coturn/coturn/blob/bc54a4d9404f7e1fee7f0f907acec5f4882b2ae1/examples/etc/turnserver.conf#L610)
  - [eturnal: enable_turn false in listen directive](https://eturnal.net/doc/)
- Make it fast and reliable:
  - webrtc connection setup time is highly dependent on how quickly we get back the required data. I vote for redundant geosteered load balancing. Perhaps starting with two zones: America, and Euro, something like:
    - stun-1.holo.host geo steered load balance = [stun-1-america.holo.host, stun-1-euro.holo.host]
    - stun-2.holo.host geo steered load balance = [stun-2-america.holo.host, stun-2-euro.holo.host]
    - then set STUN config for RTC connections to ["stun:stun-1.holo.host", "stun:stun-2.holo.host"]
    - for a total of 4 stun servers

### PR DESCRIPTION
**Pull request by steveej**
fixes https://github.com/holochain/holochain/issues/3479.

- [x] deploy to `stun-0.main.infra.holo.host`.
- [x] added alias URL `stun-1.main.infra.holo.host`
- [x] test deployed instance using `hc_service_check`

follow-up:
- [ ] set up monitoring and alerting for `stun-0.main.infra.holo.host`

### CODE REVIEW

File: modules/flake-parts/nixosConfigurations.stun-0.main.infra.holo.host/configuration.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  system.stateVersion = "23.11";
+
+  services.holochain-turn-server = {
+    enable = true;
+    url = fqdn;
+    address = ipv4;
+    listening-port = null;
+    nginx-http-port = 80;
+    username = "test";
+    credential = "test";
```
Comment (neonphog) [inline modules/flake-parts/nixosConfigurations.stun-0.main.infra.holo.host/configuration.nix]: these and the `lt-cred-mech` are irrelevant for stun, they are turn only.

File: modules/flake-parts/holochain-turn-server.nix [hunk 2]
```diff
@@ -49,6 +49,13 @@
           default = 82;
         };
 
+        listening-port = lib.mkOption {
+          description = "port for nginx to listen on for answering ACME challenges";
+          type = lib.types.nullOr lib.types.int;
+          # skipping 81 because it's the default coturn alternative http port
+          default = 80;
```
Comment (neonphog) [inline modules/flake-parts/holochain-turn-server.nix]:
```suggestion
          default = 80;
```

File: modules/flake-parts/holochain-turn-server.nix [hunk 3]
```diff
@@ -49,6 +49,13 @@
           default = 82;
         };
 
+        listening-port = lib.mkOption {
+          description = "port for nginx to listen on for answering ACME challenges";
```
Comment (neonphog) [inline modules/flake-parts/holochain-turn-server.nix]: Copy paste error?

File: modules/flake-parts/nixosConfigurations.stun-0.main.infra.holo.host/configuration.nix [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  ];
+
+  system.stateVersion = "23.11";
+
+  services.holochain-turn-server = {
+    enable = true;
+    url = fqdn;
+    address = ipv4;
+    listening-port = null;
+    nginx-http-port = 80;
```
Comment (neonphog) [inline modules/flake-parts/nixosConfigurations.stun-0.main.infra.holo.host/configuration.nix:51]: Is this correct? Wasn't the default 82? Will this conflict with the port 80 stun server?

File: modules/flake-parts/holochain-turn-server.nix (hunk 3 above)
Comment (steveej) [reply modules/flake-parts/holochain-turn-server.nix]: indeed, thanks!

File: modules/flake-parts/holochain-turn-server.nix (hunk 2 above)
Comment (steveej) [reply modules/flake-parts/holochain-turn-server.nix]: indeed, thanks!

File: modules/flake-parts/nixosConfigurations.stun-0.main.infra.holo.host/configuration.nix (hunk 4 above)
Comment (steveej) [reply modules/flake-parts/nixosConfigurations.stun-0.main.infra.holo.host/configuration.nix:51]: for this instance coturn only listens on 443. this alleviates the need for the external reverse-proxy that intercepts the ACME challenge request.

Comment (neonphog) [reply modules/flake-parts/nixosConfigurations.stun-0.main.infra.holo.host/configuration.nix:51]: I think I'm comfortable with this for the foreseeable future. The performance benefits of a port 80 stun server (if there even are any) are probably minuscule enough that they wont materially matter to us.

**Review by neonphog — APPROVED** [review]
Nice, thanks!

**Pull request merged**